### DISCUSSION TITLE
clojure macros?

### DISCUSSION THREAD

**Discussion by bhougland18** [gilch/hissp#162] [Q&A] [answered]
> It is a goal of the project to allow a more Clojure-like reader and a complete function/macro library. But while this informs the design of the compiler, it is beyond the scope of Hissp proper, and does not belong in the Hissp repository.

Does the  Clojure macro library  currently exist?

**Comment by gilch** [comment, marked as answer]
Short answer is "no".

The Clojure-like reader does have an (incomplete) prototype, ~~not yet published,~~ based on EDN. I never said that any function/macro library I write was going to copy Clojure's, only that it would be "complete" in some sense, however, it is a _design goal_ of Hissp that a fairly Clojure-like function/macro library be *possible* to write with Hissp. If you want to try writing one, I would probably add a link to it in Hissp's docs, but it doesn't belong in this repo.

Hissp's bundled macros are currently much more complete [on master](https://hissp.readthedocs.io/en/latest/hissp.macro.html#module-hissp.macros._macro_) than [on PyPI](https://hissp.readthedocs.io/en/v0.3.0/hissp.basic.html#module-hissp.basic._macro_), and are at least partially inspired by Clojure's (see the [wiki page](https://github.com/gilch/hissp/wiki/Bundled-macro-candidates#clojure) for equivalents), but with the restriction that the compiled output be standalone, i.e. it will run on Python without Hissp installed. This may already suffice for your needs.

[Hebigo's](https://github.com/gilch/hebigo) macro suite is even more complete and powerful, but without the standalone restriction. It's mostly based around Python statements, but does include a Clojure-like destructuring let.

**Reply by gilch** [reply]
UPDATE: the EDN-based prototype [has been published.](https://github.com/gilch/garden-of-edn)
